src/content/docs/cheerp/guides/clientlib.md

```diff
@@ -0,0 +1,327 @@
+and return a specific type instead.
+
+One example of this is the `client::JSON.parse` function. This function
+now returns `client::Object*`. The error you will get is slightly confusing.
+
+```
+error: expected '(' for function-style cast or type construction
+    client::Object* object = client::JSON.parse<client::Object*>(string);
+                                                ~~~~~~~~~~~~~~^
+error: expected expression
+    client::Object* object = client::JSON.parse<client::Object*>(string);
+                                                               ^
+```
+
+To fix the error, you can simply remove the type parameter. If you need a type
+that is more specific than what the function returns, you can use `->cast<T>()`
+or `->cast()`.
+
+```cpp
+    client::Object* object = client::JSON.parse(string);
+```
+
+## Redefinition of client class
+
+The old clientlib was missing some modern javascript types. To use these types,
+you may have defined them yourself in the `client` namespaces.
+
+The new clientlib already contains many of these modern types. Attempting to
+redefine them yourself results in an error.
+
+```
+error: redefinition of 'Set' as different kind of symbol
+    class Set { /* ... */ };
+          ^
+note: previous definition is here
+    class Set: public Object {
+          ^
+```
+
+To fix this, simply remove your definition of the duplicate class. The one in
+clientlib should have all the methods you need.
+
+## Renamed classes
+
+Some classes have been replaced with newer classes that have a different name.
+
+For example, `client::Element::getBoundingClientRect` now returns
+`client::DOMRect*` instead of `client::ClientRect*`.
+
+```
+error: cannot initialize a variable of type 'client::ClientRect *' with an rvalue of type 'DOMRect *'
+    client::ClientRect* rect = element->getBoundingClientRect();
+                        ^      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+```
+
+In this case, the error can be fixed by simply replacing `client::ClientRect*`
+with `client::DOMRect*`.
+
+```cpp
+    client::DOMRect* rect = element->getBoundingClientRect();
+```
+
+## Ambiguous overloads
+
+Some functions have added overloads that cause previously valid code to now be
+ambiguous. This often happens when passing an integer to a function that has
+overloads for both `double` and some pointer type, or when a function has
+overloads for `const String&` and `const _Union<T...>&`.
+
+```
+error: call to constructor of 'client::Date' is ambiguous
+    new client::Date(0);
+        ^            ~
+note: candidate constructor
+        Date(double value);
+        ^
+note: candidate constructor
+        Date(VarDate* vd);
+        ^
+```
+
+To fix the error, change the type of the argument to match the overload that
+you intend to call.
+
+```cpp
+    new client::Date(0.0); // calls double constructor
+    new client::Date(nullptr); // calls VarDate* constructor
+```
+
+## What is `_Any`, `_Union`, and `_Function`?
```

Comment:
This should be on a reference page, not a migration guide

Reply:
I started writing a reference page for these features but it was getting very long and I wasn't very happy with it. The plan is to ship the new clientlib ASAP and I think just these sections in the migration guide will be good enough for now. I'll probably be coming back to this to write more extensive documentation in the future, at which point the sections in the migration guide can be replaced with a link to the actual documentation.